src/dmd/ctfeexpr.d

```diff
@@ -1254,9 +1261,11 @@ private int ctfeRawCmp(const ref Loc loc, Expression e1, Expression e2)
     {
         real_t r1 = e1.type.isreal() ? e1.toReal() : e1.toImaginary();
         real_t r2 = e1.type.isreal() ? e2.toReal() : e2.toImaginary();
+        if (identity)
+            return !RealEquals(r1, r2);
```

Comment:
Note: RealEquals corresponds to `is` rather than `==` (all NaNs are equal, +0.0 and -0.0 are unequal), so it's correct.